src/main/java/org/elasticsearch/common/unit/TimeValue.java

```diff
@@ -285,17 +288,15 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) return false;
 
         TimeValue timeValue = (TimeValue) o;
-
-        if (duration != timeValue.duration) return false;
-        if (timeUnit != timeValue.timeUnit) return false;
-
+        if (timeUnit.toNanos(duration) != timeValue.timeUnit.toNanos(timeValue.duration)) {
```

Comment:
This, plus the final return, can be simplified to just return the value of an `==` comparison between the two values.